conformance/lang/expressions/transactional-expr/transactional_expr.balt

```diff
@@ -0,0 +1,558 @@
+            io:println(transactional);
+        }
+    };
+
+    transaction {
+        c.f1(); // output true
+        io:println(transactional); // output true
+        c->f2(); // output true
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: transactional-expr, module-type-defn, object-type, object-constructor-expr, method-call-expr,
+        check, optional-type, error, int, transaction-stmt, commit-action
+
+type ServiceObj service object {
+    transactional function f1();
+};
+
+function init() returns error? {
+    ServiceObj c = service object {
+        transactional function f1() {
+            io:println(transactional);
+        }
+    };
+
+    io:println(transactional); // output false
+    transaction {
+        c.f1(); // output true
+        io:println(transactional);
+        check commit;
+        io:println(transactional); // output false
+    }
+}
+
+Test-Case: output
+Description: Test whether the transactional expression is being evaluated in transaction mode.
+Labels: transactional-expr, check, optional-type, error, int, retry-transaction-stmt, commit-action, error:Retriable
+
+
+function init() returns error? {
+    int a = 10;
+
+    retry(3) transaction {
+        if transactional && a < 25 {
+            a += 5;
+            check trxErrorInRetry();
+        }
+
+        check commit;
+
+        if !transactional {
+            a += 20;
+        }
+    }
+
+    if !transactional {
+        a += 30;
+    }
+
+    io:println(a); // @output 75
+}
+
+function trxErrorInRetry() returns error {
+    return error error:Retriable("TransactionError");
+}
+
+Test-Case: output
+Description: Test transactional expression as expression in if statement.
+Labels: transactional-expr, if-else-stmt, compound-assignment-stmt, check, optional-type, error, int,
+        transaction-stmt, commit-action
+
+function init() returns error? {
+    int a = 10;
+
+    transaction {
+        a += f1();
+        if transactional {
+            a += 10;
+        }
+        check commit;
+        if transactional {
+            a += 20;
+        }
+    }
+
+    if transactional {
+        a += 30;
+    }
+
+    io:println(a); // @output 60
+}
+
+transactional function f1() returns int {
+    if transactional {
+        return 40;
+    }
+    return -10;
+}
+
+Test-Case: output
+Description: Test transactional expression as expression in if statement.
+Labels: transactional-expr, if-else-stmt, compound-assignment-stmt, check, optional-type, error, int,
+        transaction-stmt, commit-action, rollback-stmt
+
+function init() returns error? {
+    int a = 10;
+
+    transaction {
+        if transactional {
+            a += 5;
+            rollback;
+            if transactional {
+                a += 6;
+            }
+        } else {
+            check commit;
```

Comment:
Should this be an error? Since commit cannot be invoked outside a transactional mode

Reply:
Yeah. we can probably do that. :thinking: It would be a breaking change though 😬

Reply:
Can you create a issue please?

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/35388